app/src/main/java/de/danoeh/antennapod/fragment/FavoriteEpisodesFragment.java

```diff
@@ -58,54 +60,48 @@ public void onStop() {
     @Override
     protected void resetViewState() {
         super.resetViewState();
-        undoBarController = null;
     }
 
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
         View root = super.onCreateViewHelper(inflater, container, savedInstanceState,
-                R.layout.episodes_fragment_with_undo);
+                R.layout.all_episodes_fragment);
 
-        listView.setRemoveListener(which -> {
-            Log.d(TAG, "remove(" + which + ")");
-            if (subscription != null) {
-                subscription.unsubscribe();
+        ItemTouchHelper.SimpleCallback simpleItemTouchCallback = new ItemTouchHelper.SimpleCallback(0, ItemTouchHelper.LEFT | ItemTouchHelper.RIGHT) {
+            @Override
+            public boolean onMove(RecyclerView recyclerView, RecyclerView.ViewHolder viewHolder, RecyclerView.ViewHolder target) {
+                return false;
             }
-            FeedItem item = (FeedItem) listView.getAdapter().getItem(which);
-
-            DBWriter.removeFavoriteItem(item);
-
-            undoBarController.showUndoBar(false,
-                    getString(R.string.removed_item), new FeedItemUndoToken(item,
-                            which)
-            );
-        });
-
-        undoBarController = new UndoBarController<FeedItemUndoToken>(root.findViewById(R.id.undobar), new UndoBarController.UndoListener<FeedItemUndoToken>() {
-
-            private final Context context = getActivity();
 
             @Override
-            public void onUndo(FeedItemUndoToken token) {
-                if (token != null) {
-                    long itemId = token.getFeedItemId();
-                    DBWriter.addFavoriteItemById(itemId);
+            public void onSwiped(RecyclerView.ViewHolder viewHolder, int swipeDir) {
+                AllEpisodesRecycleAdapter.Holder holder = (AllEpisodesRecycleAdapter.Holder)viewHolder;
+                Log.d(TAG, "remove(" + holder.getItemId() + ")");
+
+                if (subscription != null) {
+                    subscription.unsubscribe();
+                }
+                FeedItem item = holder.getFeedItem();
+                if (item != null) {
+                    DBWriter.removeFavoriteItem(item);
+
+                    Snackbar snackbar = Snackbar.make(root, getString(R.string.removed_item),
+                            Snackbar.LENGTH_LONG);
+                    snackbar.setAction(getString(R.string.undo), v -> {
+                        DBWriter.addFavoriteItem(item);
```

Comment:
This is probably the solution, however we'd have to update our supportlib use to 23 to get it: http://stackoverflow.com/a/32190462/6839

(though someone else suggests a hack/workaround to use in the meantime). That could be promising. It would also get rid of the runnable business we have to do for NewEpisodesFragment.